### PR TITLE
Create project, migrate sources from runthebuilds

### PR DESCRIPTION
**Pull request by nre-ableton**
Fixes https://github.com/AbletonAppDev/devtools/issues/851, ping @AbletonDevTools/gotham-city

### CODE REVIEW

File: vars/postgres.groovy [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  // the postgres container runs into all sorts of weird problems during initialization.
+  // Also, while we're at it, we can define the POSTGRES_DB environment variable which
+  // will instruct the container to create a database for us with the given name.
+  def dockerfile = "${tempDir}/Dockerfile"
+  def postgresUser = "jenkins"
+  writeFile(
+    file:dockerfile,
+    text: """
+      FROM postgres:${postgresVersion}
+      RUN useradd --uid 1001 --user-group ${postgresUser}
```
Comment (ala-ableton) [inline vars/postgres.groovy]: @nre-ableton shouldn't we address this before moving that code in a public repository?

**Comment by nre-ableton** [comment]
@ala-ableton Pushed a fixup, it's untested but should fix the issue.

File: README.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+```groovy
+stage('Test') {
+  postgres.withDb('testdb', '9.6') {
+    sh 'run-db-tests.sh'
+  }
+}
+```
+
+The above code snippit will pull the Postgres Docker image for version 9.6, and start the
```
Comment (ala-ableton) [inline README.md]: Typo: `snippit` -> `snippet`

File: README.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+container. The commands in the closure body will be executed *outside* of the container on
+the node.
+
+The library also makes sure that the container is fully started and ready to communicate
+with before executing the closure body. To do this, it calls `pg_isready` until it
+succeeds. Although it catches and suppresses errors from the failed calls while the
+container is starting, you may see some red X's in Blue Ocean's output. They can be safely
+ignored.
```
Comment (ala-ableton) [inline README.md]: Nice!

**Comment by ala-ableton** [comment]
Once rebased, LGTM :+1:

**Comment by nre-ableton** [comment]
@ablbot autosquash

**Pull request merged**